### DISCUSSION TITLE
Sprite 1x6d (Parachute Galoomba/Bob-omb generator) info

### DISCUSSION THREAD

**Discussion by ghost** [city41/smaghetti#74] [General]
I looked into Sprite 1x6d (which you said you had trouble getting to work) and here are some things about it (some known, some probably not known):
- It is a 5-byte sprite. Graphic set is [-1, -1, 0, -1, -1, 6] though you probably don't have to set the 3rd byte if it's spawning Galoombas.
- As mentioned in TCRF, if the x-position is even (x=0), then it will spawn Parachute Galoombas (Para-Galoombas?). Odd x-positions (x=1) will spawn Parabombs.
- In order for the generator to spawn enemies, Mario must be above and to the right of it. The generator also has to be a screen away from Mario's spawn point in order for it to activate, otherwise he would have to move to the left and back.
- The fifth byte tells the sprite how many spawned enemies can be onscreen before it stops. It will continue spawning after one of those enemies either dies or leaves the screen. 5 is the value for the only used instance in the game.
- The Galoomba generator will count non-spawned Galoombas for how many are onscreen, but not Bob-ombs. Vice-versa for the Bob-omb generator.
- Sprite 1x6E, a 6-byte sprite, will stop the generator when Mario reaches it. I'm not sure how to get the generator to restart without going all the way back to its initial point. It could be related to its two bytes, but they need figuring out. This sprite seems to be unused, and I'm not sure if any other "generator stoppers" are 6-byte.
- Fun fact: the landed Bob-ombs still use set 3, even though there's graphics for them in the Galoomba/Wiggler/Parachute gfx set.

**Comment by city41** [comment]
This is great! I added ParaGaloombaBobombGenerator as an entity available in the unfinished tab.

Technically it's not unfinished, although I do want to change it to show either a galoomba or bobomb based on the current x value. But also need to think how people can use these without getting confused. It might just mean adding a documentation page on them and linking to it from the palette.

Also nice to hear about "generator stopper" I thought that might be the case those exist.

**Reply by city41** [reply]
It seems like for this generator, for it to work it must be placed 15 or more tiles to the right of Mario's starting position. This is also true of the bowser fireball generator
